### DISCUSSION TITLE
How to handle logging

### DISCUSSION THREAD

**Discussion by WillieCubed** [UTDNebula/nebula-api#26] [Ideas]
For a full 1.0 release of the Nebula API, we're going to need some ability to monitor the uptime of the API along with any errors that rise from its operation. As we're intending to use stateless containers to handle API responses, we'll be unable to use log files.

I recommend a third-party service like [Sentry](https://sentry.io/for/node/) to avoid having to easily monitor errors when they arise, triage them and link them to GitHub issues, and use a well-documented API to implement this at all levels of our stack. Additionally, something like this would allow us to centralize logging for all Project Nebula services to make maintenance simpler in the long run.

@CharlieMahana, what do you think?

**Comment by charliemahana** [comment]
This seems like a really good solution. Ignoring the cost of the service, I would be completely for it. The cost seems quite high given the limitations we are trying to place on our projected monthly cost. If this is within our budget lets go for it.

**Comment by charliemahana** [comment]
Evidently, more research into how Cloud Run works suggests that we can just use a simple STDOUT logger like [morgan](https://github.com/expressjs/morgan) since GCP with automatically route all STDOUT and STDERR logging to Cloud Logging. Given our current position, this seems like a much more economically viable option.
